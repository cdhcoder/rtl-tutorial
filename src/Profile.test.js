import React from 'react';
import { render } from '@testing-library/react';
import Profile from './Profile';

describe('<Profile />', () => {
    it('matches snapshot', () => {
        const utils = render(<Profile username="cdhcoder" name="cdh" />);
        expect(utils.container).toMatchSnapshot();
    });

    it('shows the props correctly', () => {
        const utils = render(<Profile username="cdhcoder" name="cdh" />);
        utils.getByText('cdhcoder'); // cdhcoder라는 text를 가진 element 확인
        utils.getByText('(cdh)');
    });
});