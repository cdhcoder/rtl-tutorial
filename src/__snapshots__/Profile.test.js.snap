// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      cdhcoder
    </b>
    <span>
      (
      cdh
      )
    </span>
  </div>
</div>
`;
